src/document.rs

```diff
@@ -14,10 +14,37 @@ use chrono::{offset::TimeZone, Datelike, Timelike, Utc};
 use tantivy as tv;
 
 use crate::{facet::Facet, to_pyerr};
-use pyo3::{PyMappingProtocol, PyObjectProtocol};
-use std::{collections::BTreeMap, fmt};
+use serde_json::Value as JsonValue;
+use std::{
+    collections::{BTreeMap, HashMap},
+    fmt,
+};
 use tantivy::schema::Value;
 
+fn value_to_object(val: &JsonValue, py: Python<'_>) -> PyObject {
+    match val {
+        JsonValue::Null => py.None(),
+        JsonValue::Bool(b) => b.to_object(py),
+        JsonValue::Number(n) => match n {
+            n if n.is_i64() => n.as_i64().to_object(py),
+            n if n.is_u64() => n.as_u64().to_object(py),
+            n if n.is_f64() => n.as_f64().to_object(py),
+            _ => panic!("number too large"),
```

Comment:
Sorry, I know this is slightly pedantic, but I'm pretty sure you can match on the num variant i.e `Number::I64`  which should remove the need for this panic. I think it'll be more idiomatic that way. :)

Reply:
The enum inside Number is a enum of 

``` Rust
enum N {
    PosInt(u64),
    /// Always less than zero.
    NegInt(i64),
    /// Always finite.
    Float(f64),
}

```
And it's a private field. So I don't figure out how to match them directly.

Reply:
Thank you! But I don't have the write access. Please help to merge it.

Reply:
👍